File: Leetcode/Easy/23-Best-time-to-buy-and-sell-stock/solution.py
class Solution(object):
    def maxProfit(self, prices):
        maxProf = 0
        buy = 0
        sell = 1
        while sell < len(prices):
            curProfit = prices[sell] - prices[buy]
            if prices[buy] < prices[sell]:
                maxProf = max(curProfit,maxProf)
            else:
                buy = sell
            sell += 1
        return maxProf

File: Leetcode/Easy/23-Best-time-to-buy-and-sell-stock/solution.js
var maxProfit = function(prices) {
    let left = maxGlobal = 0;
    let right = 1;
    while (right <= prices.length) {
        let currentProfit = prices[right] - prices[left];
        if (prices[left] < prices[right]) maxGlobal = Math.max(currentProfit, maxGlobal);
        else {
            left = right
        }
        right++;
    }
    return maxGlobal
};